SimplicialComplexes: Propose fix for (co-)homology

lkastner opened the pull request:
The free part of the (co-)homology is missing, since in
```
function _convert_finitely_generated_abelian_group(A::Polymake.HomologyGroupAllocated{Polymake.Integer})
    vec = ones(Int, Polymake.betti_number(A))
    torsion_i = Polymake.torsion(A)
    for (p,k) in torsion_i
        append!(vec, fill(p,k))
    end
    return vec
end
```
it is given by ones instead of zeros. Here I propose a fix by writing the resulting group as a `direct_product`. Instead `vec` could also just be initialized with zeros. Let me know which solution you prefer.

lkastner committed:
SimplicialComplexes: Propose fix for homology

fieker commented:
what ever you prefer. The _ functions are internal anyway... just be careful with direct_product as it might return maps as well

lkastner committed:
SimplicialComplexes: Add tests, repair jldoctests

lkastner committed:
DevDocs: How to autofix jldoctests

lkastner committed:
docs: Mention convenience methods for indexing switch in Polymake.jl

lkastner committed:
SimplicialComplexes: Simplify code after previous refactor

lkastner commented:
> what ever you prefer. The _ functions are internal anyway... just be careful with direct_product as it might return maps as well

Michael decided he would prefer to go with the `zeros` solution, since the maps from `direct_product` are not really the relevant ones and might lead to confusion.

lkastner commented:
On an unrelated note, I also
- added some helpful hints in the docs
- removed/simplified some codepieces, this was possible now after previous refactors.
